SonarQube.Client.Tests/Infra/MocksHelper.cs

```diff
@@ -58,18 +58,31 @@ internal static class MocksHelper
         /// </summary>
         public static void SetupHttpRequest(Mock<HttpMessageHandler> messageHandlerMock, string requestRelativePath, string response,
             HttpStatusCode statusCode = HttpStatusCode.OK, string basePath = ValidBaseAddress)
+        {
+            var responseMessage = new HttpResponseMessage
+            {
+                StatusCode = statusCode,
+                Content = new StringContent(response)
+            };
+
+            SetupHttpRequest(messageHandlerMock, requestRelativePath, responseMessage, basePath);
+        }
+
+        public static void SetupHttpRequest(Mock<HttpMessageHandler> messageHandlerMock,
```

Comment:
New overload that accepts HttpResponseMessage instead of a string, and the optional additional headers.
The existing overload calls this one.